src/python/pants/core/goals/run.py

```diff
@@ -15,6 +17,30 @@
 from pants.option.custom_types import shell_str
 from pants.option.global_options import GlobalOptions
 from pants.util.contextutil import temporary_dir
+from pants.util.frozendict import FrozenDict
+from pants.util.meta import frozen_after_init
+
+
+@frozen_after_init
+@dataclass(unsafe_hash=True)
+class RunRequest:
+    digest: Digest
+    binary_name: str
+    extra_args: Tuple[str, ...]
```

Comment:
Nit: "extra" to me tends to imply that things are added afterward. Maybe `prefix_args`?